### ISSUE TITLE
🐛 [firebase_auth] NPE: io.flutter.plugins.firebase.auth.FlutterFirebaseAuthPlugin.parseUserInfoList

### ISSUE THREAD

**Issue by rohansohonee** [firebase/flutterfire#9798]
**Crash stack trace:**
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke interface method 'java.lang.String com.google.firebase.auth.x0.v()' on a null object reference
       at io.flutter.plugins.firebase.auth.FlutterFirebaseAuthPlugin.parseUserInfoList(FlutterFirebaseAuthPlugin.java:26)
       at io.flutter.plugins.firebase.auth.FlutterFirebaseAuthPlugin.parseFirebaseUser(FlutterFirebaseAuthPlugin.java:129)
       at io.flutter.plugins.firebase.auth.IdTokenChannelStreamHandler.lambda$onListen$0(IdTokenChannelStreamHandler.java:21)
       at io.flutter.plugins.firebase.auth.IdTokenChannelStreamHandler.$r8$lambda$YFUVy4vuqZSNJHZYCVP6LRIaeoc(IdTokenChannelStreamHandler.java)
       at io.flutter.plugins.firebase.auth.IdTokenChannelStreamHandler$$InternalSyntheticLambda$3$e66cf25435250bec3a306a85fa9de6077b3eb90c107a0816a27fd84112fe3053$0.onIdTokenChanged(IdTokenChannelStreamHandler.java:6)
       at com.google.firebase.auth.zzl.run(zzl.java:52)
       at android.os.Handler.handleCallback(Handler.java:938)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8751)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
```

### Steps to reproduce
Not known. Crash was reported from firebase_crashlytics.

## Additional context

**Device details:**
```
Device
Brand:samsung
Model:Galaxy S21 5G
Orientation: Portrait
RAM free: 1.36 GB
Disk free: 14.33 GB
Operating System
Version:Android 12
Orientation: Portrait
Rooted:No
Crash
Date:Oct 25, 2022, 10:05:21 PM
```

**Pubspec.yaml:**
```
firebase_core: 1.24.0
firebase_auth: 3.11.2
firebase_crashlytics: 2.8.13
```

---

### Flutter version
```
Flutter 3.3.3 • channel stable
```

---

**Comment by danagbemava-nc** [comment]
Hi @rohansohonee, how many instances of this crash have you experienced? 

How many devices are affected? 

Can you try upgrading your firebase dependencies to the latest versions to see if you still experience this issue?

Thank you

**Comment by rohansohonee** [comment]
**Crashlytics period "Last 90 days":**
3 crash events affecting 3 users

---
**Crash 1**
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke interface method 'java.lang.String com.google.firebase.auth.x0.w()' on a null object reference
       at io.flutter.plugins.firebase.auth.FlutterFirebaseAuthPlugin.parseUserInfoList(FlutterFirebaseAuthPlugin.java:26)
       at io.flutter.plugins.firebase.auth.FlutterFirebaseAuthPlugin.parseFirebaseUser(FlutterFirebaseAuthPlugin.java:129)
       at io.flutter.plugins.firebase.auth.IdTokenChannelStreamHandler.lambda$onListen$0(IdTokenChannelStreamHandler.java:21)
       at io.flutter.plugins.firebase.auth.IdTokenChannelStreamHandler.$r8$lambda$YFUVy4vuqZSNJHZYCVP6LRIaeoc(IdTokenChannelStreamHandler.java)
       at io.flutter.plugins.firebase.auth.IdTokenChannelStreamHandler$$InternalSyntheticLambda$3$e66cf25435250bec3a306a85fa9de6077b3eb90c107a0816a27fd84112fe3053$0.onIdTokenChanged(IdTokenChannelStreamHandler.java:6)
       at com.google.firebase.auth.zzl.run(zzl.java:52)
       at android.os.Handler.handleCallback(Handler.java:790)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:7000)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:441)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1408)
```
**Device details:**
```
Device
Brand:samsung
Model:Galaxy J7 Prime
Orientation: Portrait
RAM free: 974.47 MB
Disk free: 19.71 GB
Operating System
Version:Android 8.1.0
Orientation: Portrait
Rooted:No
Crash
Date:Sep 5, 2022, 8:39:30 AM
```

**Pubspec.yaml**
```
firebase_core: 1.21.1
firebase_auth: 3.7.0
firebase_crashlytics: 2.8.9
```

---
**Crash 2**
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke interface method 'java.lang.String com.google.firebase.auth.x0.v()' on a null object reference
       at io.flutter.plugins.firebase.auth.FlutterFirebaseAuthPlugin.parseUserInfoList(FlutterFirebaseAuthPlugin.java:26)
       at io.flutter.plugins.firebase.auth.FlutterFirebaseAuthPlugin.parseFirebaseUser(FlutterFirebaseAuthPlugin.java:129)
       at io.flutter.plugins.firebase.auth.IdTokenChannelStreamHandler.lambda$onListen$0(IdTokenChannelStreamHandler.java:21)
       at io.flutter.plugins.firebase.auth.IdTokenChannelStreamHandler.$r8$lambda$YFUVy4vuqZSNJHZYCVP6LRIaeoc(IdTokenChannelStreamHandler.java)
       at io.flutter.plugins.firebase.auth.IdTokenChannelStreamHandler$$InternalSyntheticLambda$3$e66cf25435250bec3a306a85fa9de6077b3eb90c107a0816a27fd84112fe3053$0.onIdTokenChanged(IdTokenChannelStreamHandler.java:6)
       at com.google.firebase.auth.zzl.run(zzl.java:52)
       at android.os.Handler.handleCallback(Handler.java:883)
       at android.os.Handler.dispatchMessage(Handler.java:100)
       at android.os.Looper.loop(Looper.java:224)
       at android.app.ActivityThread.main(ActivityThread.java:7561)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:995)
```
**Device details:**
```
Device
Brand:Xiaomi
Model:Redmi 9
Orientation: Portrait
RAM free: 757.73 MB
Disk free: 1.54 GB
Operating System
Version:Android 10
Orientation: Portrait
Rooted:No
Crash
Date:Oct 4, 2022, 9:14:54 PM
```

**Pubspec.yaml**
```
firebase_core: 1.22.0
firebase_auth: 3.9.0
firebase_crashlytics: 2.8.10
```

---
**Crash 3**
See first comment of this issue.

**Comment by danagbemava-nc** [comment]
Labeling for further investigation. 
This might already be fixed since there was a similar issue reported a while back in https://github.com/firebase/flutterfire/issues/8517 but the crash appears to be on different lines this time. 

/cc @russellwheatley

**Comment by Lyokone** [comment]
We fixed it in #9960, it will be released this afternoon
